### DISCUSSION TITLE
One store with multiple slices

### DISCUSSION THREAD

**Discussion by utkubiz** [pmndrs/zustand#2040] [Q&A] [answered]
I have 4 slices, and I've combined them into one general store. Initially, everything was working fine, but when I tried to group these slices inside the state as shown below, the (...a) part is giving the following error. Is there anyone who can provide guidance on this?

```typescript
import { create } from "zustand";
import { AssertionState, createAssertionSlice } from "./assertion-slice";
import { RequestState, createRequestSlice } from "./request-slice";
import { ResponseState, createResponseSlice } from "./response-slice";
import { StepState, createStepSlice } from "./step-slice";

type CommonState = {
  assertion: AssertionState;
  request: RequestState;
  response: ResponseState;
  step: StepState;
};

export const useStore = create<CommonState>()((...a) => ({
  assertion: { ...createAssertionSlice(...a) },
  request: { ...createRequestSlice(...a) },
  response: { ...createResponseSlice(...a) },
  step: { ...createStepSlice(...a) },
}));
```

```
Argument of type '(partial: CommonState | Partial<CommonState> | ((state: CommonState) => CommonState | Partial<CommonState>), replace?: boolean | undefined) => void' is not assignable to parameter of type '(partial: AssertionState | Partial<AssertionState> | ((state: AssertionState) => AssertionState | Partial<...>), replace?: boolean | undefined) => void'.
  Types of parameters 'partial' and 'partial' are incompatible.
    Type 'AssertionState | Partial<AssertionState> | ((state: AssertionState) => AssertionState | Partial<AssertionState>)' is not assignable to type 'CommonState | Partial<CommonState> | ((state: CommonState) => CommonState | Partial<CommonState>)'.
      Type 'AssertionState' is not assignable to type 'CommonState | Partial<CommonState> | ((state: CommonState) => CommonState | Partial<CommonState>)'.
```

**Comment by dbritto-dev** [comment]
@utkubiz would you mind sharing your types? In a ts playground

**Reply by utkubiz** [reply]
I trimmed it but you can see the general structure and error message.

[ts playground](https://www.typescriptlang.org/play?ts=5.1.6#code/JYWwDg9gTgLgBAbzgYygUwIYzXAvnAMyghDgCIAvAVwGcYMA7AEzIG4AoUSWROAZXrYAwuizQ8hYqUq16zNu3ZoAHt3jAG2KAQzIcAQRo00sYBAYCsOBOzhwaEKlD0AuOAyogARiY65FKmoo5nQootiGxqbmfAA2wK78gmgimDDQADyRJjBmFskAfHAAvHAAFACUJUVlNnYOTokADAA07LgVHEqq0OqaJjp6cABKaACOVGh0ltiItnBOsW50UBoA5n4BPTzIIfCoaWijE1MwcQlobjMp4ZnHk9OFJeXGMFXFNXULUEvkZG0dLowACeYBwQhIIBiyWeXwwRhyeTc2Wi+SsHDs6BOdDc91O1z8XUCvWCDFCtDQAmgOFKBysGQhIChaOwBUqZTKADpuRh3p95vCorlzG4kNzOXSIgjUec9FyeVVcG1MeMHjBRXBxZKjqr8fE5eLeXgARVOuwgA)

**Reply by dbritto-dev** [reply]
@utkubiz the issue is due to typing slices with the slice type while each slide get the main type (common state)

**Reply by utkubiz** [reply]
How can I solve it? do you have any suggestions? @dbritto-dev

**Reply by dbritto-dev** [reply]
@utkubiz let me create an example for you

**Reply by dbritto-dev** [reply, marked as answer]
@utkubiz here you go -> https://www.typescriptlang.org/play?ts=5.1.6#code/JYWwDg9gTgLgBAbzgYygUwIYzXAvnAMyghDgCIAvAVwGcYMA7AEzIG4AoUSWROAZXrYAwuizQ8hYqUq16zNu3ZoAHt3jAG2KAQzIcAQRo00sYBAYCsOBOzhwaEKlD0AuOAyogARiY65FKmpwGlo6enAASmgAjlRodJbYiLZwTgA2bnRQGgDmfoowAJ5gOEIkIOaJOAC8yXYYRiYwZgxuhsamlYJoHHbosfEwblEDCd1+HOxFJfxpwHoimDDQADx8ADRwAHJ8AHxwtVWLYlBrmwDaALoX19t7AarQ8MjmdCii2O1NLXxzrrPzNDHZanMogCoWbqbL6dSFWfa1AAUAEoDvtETY7A4nP8AAzrdi4ZGTQJPFCvZ4fNAjOIJP5oNy-QHA1ZgiFVTY0wZVBFwFFovmY1JQDLkMgEokkx48F4MN60NACaA1d5LNArNldeEoxGIgB0BowqOq6KFDQ6zXMblQaphlos9P1huRBL6MVpQ1VVi5dMBTr1RolyOJ7CAA

**Reply by utkubiz** [reply]
Thank you @dbritto-dev

**Reply by dai-shi** [reply]
btw, @dbritto-dev there's a nice plugin, "Link Shortener".

<img width="379" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/59f59736-5a96-4761-8887-45f52cd680d9">
